### ISSUE TITLE
Add rule for proper indentation with preconfigured number of spaces

### ISSUE THREAD

**Issue by Johennes** [realm/SwiftLint#227]
It would be great to check for a consistent indentation using e.g. 2 or 4 space for each level of nesting. Here are some violating examples:

``` swift
class Foo {
let bar: String
}
```

``` swift
class Foo {
    func bar() {
      return "foo"
    }
}
```

In addition, it'll also be nice to check for a consistent indentation of multi-line array/dictionary or string constructs. Violating examples:

``` swift
let a = ["foo",
"foo",
"foo"]
```

``` swift
let a = "foo"
+ "bar"
+ "foobar"
```

**Comment by ecnepsnai** [comment]
Any update on this @jpsim ?  Would really value this rule.

**Comment by jpsim** [comment]
It's up for grabs, no one has started work on this as far as I know.

**Comment by arsonik** [comment]
+1

**Comment by qasim** [comment]
Working on this over at https://github.com/realm/SwiftLint/compare/master...qasim:qasim/add-indentation. ☺️

**Comment by salutis** [comment]
@qasim Any updates on this one?

**Comment by ldiqual** [comment]
@Johennes @salutis @ecnepsnai I've created a PR addressing the first part of this issue. Please feel free to 👍 if you'd like to express interest! https://github.com/realm/SwiftLint/pull/873

**Comment by fredpi** [comment]
The new `indentation_width` opt-in rule is now available (see https://github.com/realm/SwiftLint/pull/2765), so maybe we can close this issue? Or should we wait until there's also a more advanced / flexible indentation rule?

**Comment by jpsim** [comment]
Let's close this. Anyone should feel empowered to file new, more specific issues about further enhancements. Thanks @fredpi!

**Comment by keith** [comment]
FYI it looks like this rule is incompatible with this style:

```swift
        XCTAssertEqual(foo,
                       "blah")
```

which we use when the first line would overflow our column length limit

**Comment by fredpi** [comment]
@keith That's by design, as the rule suggests the following style in case of overflow:

```swift
XCTAssertEqual(
   foo,
   "blah"
)
```

**Comment by jpsim** [comment]
It' be awesome to add a configuration option to support the first style too.

**Comment by patrickbdev** [comment]
@fredpi @jpsim was support added for the first style?
It'd be useful for instances like:
```
[textField,
 callButton].forEach(addSubview)
```

**Comment by fredpi** [comment]
@patrickbdev Unfortunately, that's not yet available and currently, I don't have the time to implement it 🙁 Feel free though to open a new issue specifically explaining the desired changes!

**Comment by fredpi** [comment]
@hisavali This is expected behavior, as autocorrection is not yet available for the `indentation_width` rule. Feel free to open a dedicated issue to track progress on such a feature.
